src/main/java/com/questhelper/helpers/quests/perilousmoon/PerilousMoon.java

```diff
@@ -0,0 +1,702 @@
+		goEnterNeypotzli.addStep(inCamTorum, enterNeypotzli);
+
+		steps.put(7, goEnterNeypotzli);
+
+		ConditionalStep goTalkInNey = goEnterNeypotzli.copy();
+		goTalkInNey.addStep(and(inAntechamber, talkedToZumaInNey), talkToJessamineInNey);
+		goTalkInNey.addStep(and(inAntechamber, talkedToAttalaInNey), talkToZumaInNey);
+		goTalkInNey.addStep(inAntechamber, talkToAttalaInNey);
+		steps.put(8, goTalkInNey);
+
+		ConditionalStep goBuildPrisonCamp = goEnterNeypotzli.copy();
+		goBuildPrisonCamp.addStep(inPrison, buildPrisonCamp);
+		goBuildPrisonCamp.addStep(inAntechamber, enterPrison);
+
+		ConditionalStep goBuildEarthCamp = goEnterNeypotzli.copy();
+		goBuildEarthCamp.addStep(inEarthbound, buildEarthCamp);
+		goBuildEarthCamp.addStep(inPrison, leavePrison);
+		goBuildEarthCamp.addStep(inAntechamber, enterEarthbound);
+
+		ConditionalStep goBuildStreamCamp = goEnterNeypotzli.copy();
+		goBuildStreamCamp.addStep(inStreambound, buildStreamCamp);
+		goBuildStreamCamp.addStep(inEarthbound, leaveEarth);
+		goBuildStreamCamp.addStep(inPrison, leavePrison);
+		goBuildStreamCamp.addStep(inAntechamber, enterStreambound);
+
+		ConditionalStep buildCamps = goEnterNeypotzli.copy();
+		buildCamps.addStep(and(buildingSupplies, madePrisonCamp, madeEarthCamp), goBuildStreamCamp);
+		buildCamps.addStep(and(buildingSupplies, madePrisonCamp), goBuildEarthCamp);
+		buildCamps.addStep(buildingSupplies, goBuildPrisonCamp);
+		buildCamps.addStep(inAntechamber, takeBuildingSupplies);
+		steps.put(10, buildCamps);
+
+		ConditionalStep goReturnToJessamine = goEnterNeypotzli.copy();
+		goReturnToJessamine.addStep(inAntechamber, talkToJessamineForTalismans);
+		goReturnToJessamine.addStep(inStreambound, leaveStream);
+		goReturnToJessamine.addStep(inEarthbound, leaveEarth);
+		goReturnToJessamine.addStep(inPrison, leavePrison);
+		steps.put(12, goReturnToJessamine);
+		steps.put(13, goReturnToJessamine);
+		steps.put(14, goReturnToJessamine);
+
+		ConditionalStep goTalkToNahta = new ConditionalStep(this, enterCamTorum);
+		goTalkToNahta.addStep(inCamTorum, talkToNahta);
+		goTalkToNahta.addStep(inAntechamber, leaveNeypotzli);
+		steps.put(15, goTalkToNahta);
+
+		ConditionalStep goTalkToBlacksmith = new ConditionalStep(this, enterCamTorum);
+		goTalkToBlacksmith.addStep(inCamTorum, talkToBlacksmith);
+		goTalkToBlacksmith.addStep(inAntechamber, leaveNeypotzli);
+		steps.put(16, goTalkToBlacksmith);
+
+		ConditionalStep goFindEyatlalli = new ConditionalStep(this, enterCamTorum);
+		goFindEyatlalli.addStep(eyatlalliNearby, talkToEyatlalli);
+		goFindEyatlalli.addStep(inNeypotzli, useTalismans);
+		goFindEyatlalli.addStep(inCamTorum, enterNeypotzliAfterBlacksmith);
+		steps.put(17, goFindEyatlalli);
+		steps.put(18, goFindEyatlalli);
+
+		ConditionalStep goTalkToJessamineAfterEyatlalli = new ConditionalStep(this, enterCamTorum);
+		goTalkToJessamineAfterEyatlalli.addStep(inNeypotzli, talkToJessamineAfterEyatlalli);
```

Comment:
It would be a nice add, but not necessary, to add checks for if the user is in one of the areas & lead them to Jessamine from there